### PR TITLE
feat:support modify app files via Clusterfile

### PR DESCRIPTION
**Pull request by kakaZhou719**
### Describe what this PR does / why we need it

Proposal see: https://github.com/sealerio/sealer/issues/1969

```yaml
apiVersion: sealer.io/v2
kind: Cluster
metadata:
  creationTimestamp: null
  name: my-cluster
spec:
  hosts:
    - ips:
        - 172.16.26.162
      roles:
        - master
      ssh: {}
  image: abc:v1
  ssh:
    passwd: xxxxxx
    pk: /root/.ssh/id_rsa
    port: "22"
    user: root
status: {}
---
apiVersion: sealer.io/v2
kind: Application
metadata:
  name: my-apps
spec:
  launchApps:
    - nginx
    - yamlapp
    - mixedapp
  configs:
    - name: yamlapp
      files:
        - path: abc.yaml
          strategy: "overwrite"
          data: |
            i am raw test1
            i am raw test2
        - path: merge.yaml
          strategy: "merge"
          data: |
            data:
              test-key: test-key
            metadata:
               namespace: test-namespace     
    - name: mixedapp
      files:
        - path: rediscert
          strategy: "overwrite"
          data: |
            redis-user: root
            redis-passwd: xxx



```

```shell
[root@iZbp1chh98quny8r8r71bhZ]# cat merge.yaml 
apiVersion: v1
data:
  key1: myConfigMap1
kind: ConfigMap
metadata:
  name: myConfigMap1
---
apiVersion: v1
data:
  key2: myConfigMap2
kind: ConfigMap
metadata:
  name: myConfigMap2
```

after run merge

```shell
[root@iZbp1chh98quny8r8r71bhZ]# cat /var/lib/sealer/data/my-cluster/rootfs/application/apps/yamlapp/merge.yaml 
apiVersion: v1
data:
    key1: myConfigMap1
    test-key: test-key
kind: ConfigMap
metadata:
    name: myConfigMap1
    namespace: test-namespace
---
apiVersion: v1
data:
    key2: myConfigMap2
    test-key: test-key
kind: ConfigMap
metadata:
    name: myConfigMap2
    namespace: test-namespace
```

after overwite

```shell
[root@iZbp1chh98quny8r8r71bhZ]# cat /var/lib/sealer/data/my-cluster/rootfs/application/apps/mixedapp/rediscert 
redis-user: root
redis-passwd: xxx
```

### CODE REVIEW

**Comment by Starnop** [comment]
update files.filePath to files.path

File: cmd/sealer/cmd/cluster/run.go [hunk 1]
```diff
@@ -252,7 +263,7 @@ func createNewCluster(imageEngine imageengine.Interface, cf clusterfile.Interfac
 		}
 	}()
 
-	distributor, err := imagedistributor.NewScpDistributor(imageMountInfo, infraDriver, cf.GetConfigs())
+	distributor, err := imagedistributor.NewScpDistributor(imageMountInfo, infraDriver, cf.GetConfigs(), v2App)
```
Comment (Starnop) [inline cmd/sealer/cmd/cluster/run.go:266]: do not rely on the app for ScpDistributor

File: cmd/sealer/cmd/cluster/run.go [hunk 2]
```diff
@@ -397,6 +410,13 @@ func installApplication(appImageName string, envs []string, app *v2.Application,
 		}
 	}()
 
+	for _, info := range imageMountInfo {
+		err = v2App.FileProcess(info.MountDir)
+		if err != nil {
+			return err
```
Comment (Starnop) [inline cmd/sealer/cmd/cluster/run.go:416]: use errors.Wrapf(err,"xxx") here and add more info to facilitate the precise location of the problem

File: pkg/application/v2app.go [hunk 3]
```diff
@@ -149,3 +186,64 @@ func makeItDir(str string) string {
 	}
 	return str
 }
+
+func newFileProcessor(appFile v2.AppFile) (FileProcessor, error) {
+	switch appFile.Strategy {
+	case v2.OverWriteStrategy:
```
Comment (Starnop) [inline pkg/application/v2app.go:192]: so the `merge` strategy is also not currently supported?

File: utils/strings/strings.go [hunk 4]
```diff
@@ -158,28 +158,30 @@ func Merge(ms ...[]string) []string {
...
 }
 
-// ConvertToMap :if env list containers Unicode punctuation character,will ignore this element.
-func ConvertToMap(env []string) map[string]string {
-	envs := map[string]string{}
-	var k, v string
-	for _, e := range env {
-		if e == "" {
+// ConvertEnv Convert []string to map[string]interface{}
+func ConvertEnv(envList []string) (env map[string]interface{}) {
```
Comment (Starnop) [inline utils/strings/strings.go:162]: In my option, as common utils with package string, it's weird to see the word env here.
Would a generic field like string `stringSlice`, `stringMap`,`stringInterfaceMap`, and so on be more appropriate?

File: pkg/imagedistributor/interface.go [hunk 5]
```diff
@@ -22,7 +22,7 @@ import (
 
 type Distributor interface {
 	// Distribute each files under mounted cluster image directory to target hosts.
-	Distribute(hosts []net.IP, dest string) error
+	Distribute() error
```
Comment (Starnop) [inline pkg/imagedistributor/interface.go:25]: Does this change apply to the other two methods as well？

Comment (kakaZhou719) [reply pkg/imagedistributor/interface.go:25]: not changes, hosts IP is from ClusterImageMountInfo.

**Review by Starnop — APPROVED** [review]
LGTM

**Pull request merged**